frontend/src/context/GamemodeContext.jsx

```diff
@@ -44,12 +73,20 @@ export const Context = createContext();
 // Provider Component
 export const Provider = ({ children }) => {
     const [state, dispatch] = useReducer(Reducer, initialState);
+    const [timer, setTimer] = useState(3);
```

Comment:
same as above, this timer variable should just be a variable in the provider with its own state and dispatch call, I would just remove this entirely until we actually start using it...